SimCore/python/generators.py

```diff
@@ -311,6 +311,28 @@ def single_8gev_e_upstream_tagger():
     )
 
 
+def single_8gev_e_upstream_magnet_setEnergy(ene=8.0): 
```

Comment:
I like this idea and I'm just trying to think of a better name for this function.

What could we call upstream of the tagger? The magnet and the tagger overlap in z so upstream of the magnet is kinda the same place as upstream of the tagger. Maybe `beam_pipe` instead of `upstream_magnet` to try to say we are modeling the electron arriving infront of everything?

I would also remove the `_setEnergy` suffix, I think the argument makes the point well enough.

```suggestion
def single_8gev_beam_pipe(ene = 8.0):
```

Reply:
I dont think it should include the `_8gev_` either, since if `ene = 4.0` then this is not an 8 GeV beam. The `_e_` does make sense tho, right? i.e. it should start with `single_e`